src/apiClient.ts

```diff
@@ -6,30 +6,55 @@ export interface ApiClientConfig {
   optionServiceUrl?: string
   analyticsServiceUrl?: string
   userNotificationServiceUrl?: string
-  tokenRefreshServiceUrl?: string
   debug?: boolean
 }
 
-export interface Tokens {
-  accessToken?: string
-  refreshToken?: string
-}
-
 export interface Handlers {
   onAccessTokenUpdate?: (token: string) => void
   onFailedTokenRefresh?: () => void
 }
 
 class ApiClient {
-  static instance: ApiClient
+  private static instance: ApiClient
 
-  configuration: ApiClientConfig = {}
-  tokens: Tokens = {}
-  handlers: Handlers = {}
-  version = "v1"
+  configuration: ApiClientConfig
+  accessToken: {
+    token: string | null
+    expires: Date | null
+  }
+  handlers: Handlers
+  version: string
+  runsInBrowser: boolean
+  refreshToken: () => Promise<{ accessToken: string }>
+  ensureTokenFreshness: () => void
 
   constructor() {
+    if (ApiClient.instance) return ApiClient.instance
+
+    this.configuration = {}
+    this.accessToken = {
+      expires: null,
+      token: null,
+    }
+    this.handlers = {}
+    this.version = "v1"
+    this.runsInBrowser = typeof window !== "undefined"
```

Comment:
```suggestion
    this.hasWindowObject = typeof window !== "undefined"
```

if you want to harmonize with the naming in the deviceHelper in listings